### PR TITLE
golang filter: config parser could return error when parse failed.

### LINKED ISSUE #26232: logging / error handling in golang filters
*Title*: *How to handle errors or emit logs in golang filters*

*Description*:
Is there a standard way to handle/log errors in the filter `ConfigFactory` method?
https://github.com/envoyproxy/envoy/blob/main/examples/golang/simple/config.go#L24

I couldn't find any decent way to write logs or handle errors in the golang filter example.

### PR DESCRIPTION
**Pull request by doujiang24**
Commit Message:
fix #26232 
Also, use config parser in the example golang filter.

Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by doujiang24** [comment]
/review @wangfakang @phlax 
Thank you!

File: contrib/golang/filters/http/source/cgo.cc [hunk 1]
```diff
@@ -153,12 +153,11 @@ CAPIStatus envoyGoFilterHttpGetIntegerValue(void* r, int id, void* value) {
   });
 }
 
-CAPIStatus envoyGoFilterHttpLog(void* r, uint32_t level, void* message) {
-  return envoyGoFilterHandlerWrapper(
-      r, [level, message](std::shared_ptr<Filter>& filter) -> CAPIStatus {
-        auto mesg = referGoString(message);
-        return filter->log(level, mesg);
-      });
+static FilterLogger logger;
```
Comment (mattklein123) [inline contrib/golang/filters/http/source/cgo.cc:156]: Please do not use any raw statics do to init fiasco. See CONSTRUCT_ON_FIRST_USE. Please fix other cases as well.

**Review by mattklein123 — CHANGES_REQUESTED** [review]
LGTM with small comment.

/wait

File: contrib/golang/filters/http/source/go/pkg/http/config.go [hunk 2]
```diff
@@ -55,7 +56,14 @@ func envoyGoFilterNewHttpPluginConfig(configPtr uint64, configLen uint64) uint64
 
 	configNum := atomic.AddUint64(&configNumGenerator, 1)
 	if httpFilterConfigParser != nil {
-		configCache.Store(configNum, httpFilterConfigParser.Parse(&any))
+		id, err := httpFilterConfigParser.Parse(&any)
```
Comment (wangfakang) [inline contrib/golang/filters/http/source/go/pkg/http/config.go:59]: It's better to change the variable name `id --> parsedConfig`, as it does not return a typical ID.

**Review by wangfakang — COMMENTED** [review]
Except for some minor comments, others LGTM.

File: contrib/golang/filters/http/source/go/pkg/http/config.go (hunk 2 above)
Comment (doujiang24) [reply contrib/golang/filters/http/source/go/pkg/http/config.go]: oh, nice catch, done, thanks.

File: contrib/golang/filters/http/source/cgo.cc (hunk 1 above)
Comment (doujiang24) [reply contrib/golang/filters/http/source/cgo.cc]: Done, thanks.

**Review by mattklein123 — APPROVED** [review]

**Comment by mattklein123** [comment]
Looks like there is a merge conflict. Can you merge main?

/wait

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**